ui/src/components/forms/SegmentsPicker.tsx

```diff
@@ -0,0 +1,116 @@
+import { MinusSmallIcon } from '@heroicons/react/24/outline';
+import { useRef } from 'react';
+import Combobox from '~/components/forms/Combobox';
+import { FilterableSegment, ISegment } from '~/types/Segment';
+import { truncateKey } from '~/utils/helpers';
+
+type SegmentPickerProps = {
+  segments: ISegment[];
+  selectedSegments: FilterableSegment[];
+  segmentAdd: (segment: FilterableSegment) => void;
+  segmentReplace: (index: number, segment: FilterableSegment) => void;
+  segmentRemove: (index: number) => void;
+};
+
+export default function SegmentsPicker(props: SegmentPickerProps) {
+  const {
+    segmentAdd,
+    selectedSegments: parentSegments,
+    segments,
+    segmentRemove,
+    segmentReplace
+  } = props;
+
+  const segmentsSet = useRef<Set<string>>(new Set<string>());
```

Comment:
@markphelps Hm maybe so. I was reading up on the differences before I used `useRef`, but I believe you use `useState` for things that go into rendering and re-rendering. Since this isn't something that is being rendered onto the screen but a local instance variable it could be used with `useRef` instead?

Reply:
Yup upon further reading I think you are right. My b. Haven't used refs as much as state